test/test_linalg.py

```diff
@@ -5689,28 +5689,87 @@ def run_test(matsize, batchdims, mat_chars):
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    @dtypes(torch.double)
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
     def test_cholesky_inverse(self, device, dtype):
-        from torch.testing._internal.common_utils import random_symmetric_pd_matrix
-        a = random_symmetric_pd_matrix(5, dtype=dtype, device=device)
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test(shape, batch, upper, contiguous):
+            A = random_hermitian_pd_matrix(shape, *batch, dtype=dtype, device=device)
+            if A.numel() > 0 and not contiguous:
+                A = A.transpose(-2, -1)
+                self.assertFalse(A.is_contiguous())
+            L = torch.linalg.cholesky(A)
+            expected_inverse = torch.inverse(A)
+            L = L.conj().transpose(-2, -1) if upper else L
+            actual_inverse = torch.cholesky_inverse(L, upper)
+            self.assertEqual(actual_inverse, expected_inverse)
+
+        shapes = (0, 3, 5)
+        batches = ((), (0,), (3, ), (2, 2))
+        for shape, batch, upper, contiguous in list(itertools.product(shapes, batches, (True, False), (True, False))):
+            run_test(shape, batch, upper, contiguous)
+
+        # check the out= variant
+        A = random_hermitian_pd_matrix(3, 2, dtype=dtype, device=device)
+        L = torch.linalg.cholesky(A)
+
+        # There are two code paths currently for the out= variant
+        # 1. When 'out' tensor is in Fortran (column-major) memory format
+        # then the fast route is taken and the storage is reused directly in the computations
+        # 2. When 'out' tensor is not in Fortran format then a temporary tensor is allocated internally
+        # and the result is copied from the temporary tensor to 'out' tensor
+
+        # This test checks the first code path
+        out = torch.empty_like(A)
+        out_t = out.transpose(-2, -1).clone(memory_format=torch.contiguous_format)
+        out = out_t.transpose(-2, -1)
+        ans = torch.cholesky_inverse(L, out=out)
+        self.assertEqual(ans, out)
+        expected = torch.inverse(A)
+        self.assertEqual(expected, out)
+
+        # This test checks the second code path
+        out = torch.empty_like(A)
+        ans = torch.cholesky_inverse(L, out=out)
+        self.assertEqual(ans, out)
+        expected = torch.inverse(A)
+        self.assertEqual(expected, out)
 
-        # compute inverse directly
-        inv0 = torch.inverse(a)
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_cholesky_inverse_errors_and_warnings(self, device, dtype):
+        # cholesky_inverse requires the input to be at least 2 dimensional tensor
+        a = torch.randn(2, device=device, dtype=dtype)
+        with self.assertRaisesRegex(RuntimeError, "must have at least 2 dimensions"):
```

Comment:
I didn't update the documentation as part of this PR. I thought it might be better to revisit it later together with all other linear algebra functions and make the notation, terminology, batching support, etc. consistent then.
Documentations says 2-D tensor because the original TH-based implementation did not support a batch of matrices as an input.

Reply:
@IvanYashchuk that makes sense. Although, I would still insist that we either disallow non-2D inputs or update the documentation just to be extra careful. it shouldn't take too long!

Reply:
We will definitely update the documentation ahead of the 1.8 release. If this PR is otherwise correct we may want to do it in a follow-up PR, however.